WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift

```diff
@@ -70,6 +70,18 @@ final class HubMenuViewModel: ObservableObject {
     private let cardPresentPaymentsOnboarding: CardPresentPaymentsOnboardingUseCaseProtocol
     private let posEligibilityChecker: POSEligibilityCheckerProtocol
 
+    // TODO:
+    // Is this the right place to instantiate the product provider and use property injection?
+    lazy var posItemProvider: POSItemProvider = {
+        let storageManager = ServiceLocator.storageManager
+        let siteID = ServiceLocator.stores.sessionManager.defaultSite?.siteID ?? 0
```

Comment:
❓  is the siteID from ServiceLocator instead of `HubMenuViewModel.siteID` because of the issue in p1717043548835669/1716913986.223809-slack-CGPNUU63E where the `HubMenuViewModel.siteID` isn't accurate after switching stores? I think if `HubMenuViewModel.siteID` is indeed inaccurate, it's worth creating 
an issue as it might cause other issues within the Menu.

Reply:
That's a good question that I haven't had the chance to dig much into it yet. I wouldn't expect that those ID's are different, but they could be 🤔  specially since our `pos` dependencies are created as lazy through the lifecycle of a different component that relies its recreation directly on site switching. I've logged this separately as definitely needs more testing: https://github.com/woocommerce/woocommerce-ios/issues/12919